### PR TITLE
Clarify + test content negotiation

### PR DESCRIPTION
**Pull request by fenekku**
Something weird happened with the other branch. This is one has the cherry-picked work.

### CODE REVIEW

File: tests/test_content_negotiation.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+# NOTE: By scoping down we remove the need to check for HTTP method
+def test_choose_provided_and_accepted_mimetype():
+    # Should choose mimetype that is accepted by client and served by server
+    server_mimetypes = ['application/json', 'application/marcxml+xml']
+    client_mimetypes = parse_accept_header(
+        'text/plain,application/json,*/*',
+        MIMEAccept
+    )
+
+    mime_type = ContentNegotiator.match_by_accept(
```
Comment (lnielsen) [inline tests/test_content_negotiation.py:25]: comment: style-wise I personally prefer shorter test code like this

```python
assert 'application/json' == ContentNegotiator.match_by_accept(...)
```

File: flask_resources/content_negotiation.py [hunk 2]
```diff
@@ -18,55 +18,57 @@ class ContentNegotiator(object):
     """
 
     @classmethod
-    def match(cls, mimetypes, accept_mimetypes, formats_map, format,
+    def match(cls, mimetypes, accept_mimetypes, formats_map, fmt,
```
[the remark this answers is not in the document]
Comment (fenekku) [reply flask_resources/content_negotiation.py:21]: VSCode syntax highlighted it as reserved even if it is not :man_shrugging: . I can probably revert.

File: flask_resources/content_negotiation.py [hunk 3]
```diff
@@ -18,55 +18,57 @@ class ContentNegotiator(object):
...
-                    best_quality = quality
-                    best = m
-
-        # If no match found, but wildcard exists, them use default media
-        # type.
-        if best is None and has_wildcard:
-            best = default
-
-        return best
+        # TODO: What happens if server provides */* ? Is that possible?
```
Comment (fenekku) [inline flask_resources/content_negotiation.py:67]: From discussion: not possible (maybe assert for it)

File: tests/test_content_negotiation.py (hunk 1 above)
Comment (fenekku) [reply tests/test_content_negotiation.py:25]: Sure. Makes sense here.

File: flask_resources/content_negotiation.py [hunk 4]
```diff
@@ -18,55 +18,57 @@ class ContentNegotiator(object):
...
+              the server serves that specific mimetype, then favour
+              that mimetype no matter its quality over the wildcard.
+
+        TODO: Choose the explicit policy for this "best match".
+        Meanwhile, the policy implemented below is:
+
+        :param mimetypes: Iterable of available MIME types.
+        :param accept_mimetypes: The client's "Accept" header as MIMEAccept
+            object.
+        :param mimetypes: Default MIMEtype if wildcard received.
```
[the remark this answers is not in the document]
Comment (fenekku) [reply flask_resources/content_negotiation.py:50]: Thanks! Changed it.

File: tests/test_content_negotiation.py [hunk 5]
```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+# Copyright (C) 2020 Northwestern University.
+#
+# Flask--Resources is free software; you can redistribute it and/or modify
```
[the remark this answers is not in the document]
Comment (fenekku) [reply tests/test_content_negotiation.py:6]: Yeah, that was weird...

**Pull request merged**